Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h

```diff
@@ -368,20 +368,53 @@ namespace AZ
         //! @return True if the command line argument could be parsed, otherwise false.
         virtual bool MergeCommandLineArgument(AZStd::string_view argument, AZStd::string_view anchorKey = "",
             const CommandLineArgumentSettings& commandLineSettings = {}) = 0;
+
+        //! List of possible return codes from MergeSettings/MergeSettingsFolder
+        //! The unset value is always 0
+        //! Failure values should be < 0 and success values > 0
+        enum class MergeSettingsReturnCode
+        {
+            Unset = 0,
+            Success = 1,
+            PartialSuccess,
+            Failure = -1,
+        };
+
+        //! Encapsulates the result of a JSON Patch or JSON Merge Patch opreations
+        //! into the Settings Registry
+        struct MergeSettingsResult
+        {
+            explicit operator bool() const;
+
+            // Combine MergeSettingsResult together by concatenating the
+            // operation messages and updating the return code based on the
+            // success and failure enum values of both results
+            // Chaining can be performed on a non-const lvalue reference
+            MergeSettingsResult& Combine(MergeSettingsResult otherResult) &;
+            // rvalue reference of combine returns a copy
+            MergeSettingsResult Combine(MergeSettingsResult otherResult) &&;
+
+            // Return a reference to the operation messages string
+            const AZStd::string& GetMessages() const;
+
+            MergeSettingsReturnCode m_returnCode{ MergeSettingsReturnCode::Unset };
+            AZStd::string m_operationMessages;
+        };
+
         //! Merges the json data provided into the settings registry.
         //! @param data The json data stored in a string.
         //! @param format The format of the provided data.
         //! @param anchorKey The key where the merged json content will be anchored under.
         //! @return True if the data was successfully merged, otherwise false.
-        virtual bool MergeSettings(AZStd::string_view data, Format format, AZStd::string_view anchorKey = "") = 0;
+        virtual MergeSettingsResult MergeSettings(AZStd::string_view data, Format format, AZStd::string_view anchorKey = "") = 0;
```

Comment:
if this is a breaking API change, can you bump the `api_versions -> framework` version in `engine.json`